src/api/core/organizations.rs

```diff
@@ -929,11 +930,49 @@ async fn send_invite(org_id: &str, data: Json<InviteData>, headers: AdminHeaders
             }
         };
 
-        let mut new_user = UserOrganization::new(user.uuid.clone(), String::from(org_id));
+        let mut new_member = UserOrganization::new(user.uuid.clone(), String::from(org_id));
         let access_all = data.access_all;
-        new_user.access_all = access_all;
-        new_user.atype = new_type;
-        new_user.status = user_org_status;
+        new_member.access_all = access_all;
+        new_member.atype = new_type;
+        new_member.status = user_org_status;
+        new_member.save(&mut conn).await?;
+
+        if CONFIG.mail_enabled() {
+            let org_name = match Organization::find_by_uuid(org_id, &mut conn).await {
+                Some(org) => org.name,
+                None => err!("Error looking up organization"),
+            };
+
+            if let Err(e) = mail::send_invite(
+                &user,
+                Some(String::from(org_id)),
+                Some(new_member.uuid.clone()),
+                &org_name,
+                Some(headers.user.email.clone()),
+            )
+            .await
+            {
+                // Upon error delete the user, invite and org member records when needed
+                if user_created {
+                    Invitation::take(email, &mut conn).await;
+                    user.delete(&mut conn).await?;
+                }
+                new_member.delete(&mut conn).await?;
```

Comment:
I would either call `new_member.delete()` before the `if` statement or only in an `else` block because `user.delete()` should also delete the `UserOrganization` and even if it works it feels wrong to call `update_uuid_revision` after deleting the `User`  https://github.com/dani-garcia/vaultwarden/blob/ef4bff09eb4414117cd8ea25c809988872e2016b/src/db/models/organization.rs#L708-L709

Reply:
Oh, and Invitation::take() is also called by user.delete, so this probably can be simplified to just 
```suggestion
                if user_created {
                    user.delete(&mut conn).await?;
                } else {
                    new_member.delete(&mut conn).await?;
                }
```

Reply:
Uch, I should get some more sleep

Reply:
Besides all that, Invites are also only created if smtp is **not** enabled 🤣 , so that would have resulted in a useless query anyway. While it still is done upon deleting the user, it's probably the better option.